### PR TITLE
Update CI configs to v0.4.7

### PR DESCRIPTION
**Pull request by pionbot**
Update lint scripts and CI configs.
Excluding codecov.yml.

### CODE REVIEW

**Comment by scorpionknifes** [comment]
I am taking this one! 2nd to last let's go.

**Comment by Sean-Der** [comment]
@scorpionknifes I am starting with pion/webrtc now. I can't believe we are almost done O_O

Thank you so much for your help on this. I couldn't have done it without you.

File: client.go [hunk 1]
```diff
@@ -509,7 +513,7 @@ type buffer struct {
 	buf []byte
 }
 
-var bufferPool = &sync.Pool{
+var bufferPool = &sync.Pool{ // nolint:gochecknoglobals
```
Comment (scorpionknifes) [inline client.go:516]: sync.Pool doesn't seem suitable inside a func

File: cmd/stun-bench/main.go [hunk 2]
```diff
@@ -19,17 +20,17 @@ import (
...
-	memProfile = flag.String("memprofile", "", "file output of pprof memory profile")
-	realRand   = flag.Bool("crypt", false, "use crypto/rand as random source")
+	workers    = flag.Int("w", runtime.GOMAXPROCS(0), "concurrent workers")           // nolint:gochecknoglobals
+	addr       = flag.String("addr", "localhost", "target address")                   // nolint:gochecknoglobals
+	port       = flag.Int("port", stun.DefaultPort, "target port")                    // nolint:gochecknoglobals
+	duration   = flag.Duration("d", time.Minute, "benchmark duration")                // nolint:gochecknoglobals
+	network    = flag.String("net", "udp", "protocol to use (udp, tcp)")              // nolint:gochecknoglobals
+	cpuProfile = flag.String("cpuprofile", "", "file output of pprof cpu profile")    // nolint:gochecknoglobals
+	memProfile = flag.String("memprofile", "", "file output of pprof memory profile") // nolint:gochecknoglobals
+	realRand   = flag.Bool("crypt", false, "use crypto/rand as random source")        // nolint:gochecknoglobals
```
Comment (scorpionknifes) [inline cmd/stun-bench/main.go:30]: Keeping all flag as global var.

**Review by Sean-Der — APPROVED** [review]

File: attributes.go [hunk 3]
```diff
@@ -111,46 +111,48 @@ func (t AttrType) Value() uint16 {
 	return uint16(t)
 }
 
-var attrNames = map[AttrType]string{
```
Comment (ernado) [inline attributes.go:114]: Why this change was made?
Global readonly maps are 100% ok.

Comment (scorpionknifes) [reply attributes.go:114]: Changed to pass `gochecknoglobals`. Wrapped around a func to make it read-only.

Not sure if global map is read-only.

This also occurs for `methodName`

If you think var is better we can make a `//nolint` exception

File: fuzz.go [hunk 4]
```diff
@@ -22,16 +21,16 @@ func FuzzMessage(data []byte) int {
 	}
 	m2 := New()
 	if _, err := m2.Write(m.Raw); err != nil {
-		panic(err) // nolint
+		fatal(err)
```
Comment (at-wat) [inline fuzz.go:24]: Fuzz test is not working due to this.
